server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreMetadataLockManager.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.lockmanager;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.opensearch.index.store.RemoteDirectory;
+
+import java.io.IOException;
+import java.util.Collection;
+
+/**
+ * A Class that implements Remote Store Lock Manager by creating lock files.
+ * @opensearch.internal
+ */
+public class RemoteStoreMetadataLockManager implements RemoteStoreLockManager {
+    private static final Logger logger = LogManager.getLogger(RemoteStoreMetadataLockManager.class);
+    private final RemoteDirectory lockDirectory;
+    public RemoteStoreMetadataLockManager(RemoteDirectory lockDirectory) {
+        this.lockDirectory = lockDirectory;
+    }
+
+    @Override
+    public void acquire(LockInfo lockInfo) throws IOException {
+        assert lockInfo instanceof ShardLockInfo : "lockinfo should be instance of ShardLockInfo";
+        assert ((ShardLockInfo) lockInfo).getFileToLock() != null : "file to be locked should be provided while " +
+            "acquiring lock";
+        assert ((ShardLockInfo) lockInfo).getResourceId() != null : "resource Id should be provided while " +
+            "acquiring lock";
+        assert ((ShardLockInfo) lockInfo).getExpiryTime() != null : "expiry time for lock should be provided while " +
+            "acquiring lock";
+        try (IndexOutput indexOutput = lockDirectory.createOutput(lockInfo.getLockName(), IOContext.DEFAULT)) {
+            lockInfo.writeLockContent(indexOutput);
+        }
+    }
+
+    @Override
+    public void release(LockInfo lockInfo) throws IOException {
+        assert lockInfo instanceof ShardLockInfo : "lockinfo should be instance of ShardLockInfo";
+        String resourceId = ((ShardLockInfo) lockInfo).getResourceId();
+        assert resourceId != null : "resourceId for which we need to release lock should not be null";
+        String[] lockFiles = lockDirectory.listAll();
```

Comment:
Can we filter it based on the prefix?

Reply:
But we can still filter on prefix which is`fileToLock`, right? 

For now, we are assuming only snapshot will be acquiring the locks. Locking mechanism does not have such restriction (which is a good thing).

Reply:
the problem is to release a lock, we will not be having fileToLock. we will just be having the resourceId which acquired the lock.

For ex. in snapshot usecase, fileToLock will be the metadata file, as we are not passing this to snapshot, snapshot will only be having resourceID, which is the snapshotID.
so we are pulling all the lock files for a given resource.